dotcom-rendering/src/web/components/EnhancePinnedPost.importable.tsx

```diff
@@ -43,14 +44,19 @@ function scrollOnCollapse() {
 }
 
 export const EnhancePinnedPost = () => {
+	const [hasBeenSeen, setHasBeenSeen] = useState(false);
+	const [isInView] = useIsInView({
+		threshold: 0.1,
+		repeat: true,
+		node: pinnedPost ?? undefined,
+	});
+
 	const contentFitsContainer =
 		pinnedPostContent &&
 		pinnedPostContent.scrollHeight <= pinnedPostContent.clientHeight;
 
 	if (contentFitsContainer) hideShowMore();
 
-	const hasBeenSeen = useRef(false);
-
 	const handleClickTracking = () => {
```

Comment:
This function will get recreated on every render. That's usually okay if it is not too expensive to create, the re-renders are limited and the function is not required to be stable.

iiuc this function doesn't use any dependencies so it could it be moved inside the mount useEffect where it is used?

Reply:
I actually think this function should be moved outside of the component entirely, next to `scrollOnCollapse` and the like.

Reply:
That would also work 👍

Reply:
Thanks both for pointing this out. I moved the click handler out of the component :)